server/index.js

```diff
@@ -33,6 +33,25 @@ app.get('/api/products', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/products/:productId', (req, res, next) => {
+  const sql = `
+    select *
+      from "products"
+     where "productId" = $1
+  `;
+  const params = [req.params.productId];
+
+  db.query(sql, params)
+    .then(result => {
+      const product = result.rows[0];
+      if (!product) {
+        next(new ClientError(`${params[0]} is not a valid productId`, 404));
+      }
```

Comment:
Be sure to `return` or include an `else` block to prevent the server from attempting to respond twice. `next()` does not terminate this function.